File: auth2/auth2-demo/src/main/java/com/cwd/auth2/controller/UserController.java
package com.cwd.auth2.controller;

import com.cwd.auth2.model.User;
import com.cwd.auth2.service.UserService;
import io.swagger.annotations.Api;
import io.swagger.annotations.ApiOperation;
import io.swagger.annotations.ApiParam;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.access.prepost.PreAuthorize;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

/**
 * Created by Administrator on 2017/1/11.
 */
@Api(description = "用户管理")
@RestController
@RequestMapping(value = "/user")
public class UserController {
    @Autowired
    private UserService userService;

    @ApiOperation(value = "获取用户信息")
    @RequestMapping(value = "/getUserInfo", method = RequestMethod.GET)
    @PreAuthorize("hasRole('USER')")
    public User getUserInfo(
            @ApiParam(name = "id", value = "用户id")
            @RequestParam(value = "id", required = true) String id) {
        User userInfo = new User();
        return userInfo;
    }

}

File: sms/sms-bootstrap/src/main/scala/com/yihu/cwd/controller/UserController.scala
package com.yihu.cwd.controller

import com.yihu.cwd.service.{MenuJavaService, UserService}
import com.yihu.cwd.service.system.{RoleService, MenuService}
import io.swagger.annotations.{ApiParam, Api}
import org.json.JSONObject
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.{RequestParam, RequestMethod, RequestMapping, RestController}

/**
  * Created by chenweida on 2017/3/28.
  */
@RestController
@RequestMapping(value = Array("/user"))
@Api(description = "用户")
class UserController extends BaseController{
  @Autowired
  private var userService: UserService = _;
  @Autowired
  private var menuService:MenuService=_;
  @Autowired
  private var roleService:RoleService=_;
  @Autowired
  private var menuJavaService:MenuJavaService=_;

  @RequestMapping(value = Array("/getUserInfo"),method = Array(RequestMethod.GET))
  def getUserInfo(
                   @ApiParam(name = "userId", value = "用户id", required = true) @RequestParam(value = "userId", required = true)userId:String):String={
    try{
      //根据用户id找到用户信息
      var user=userService.findUserById(userId)
      //找到用户的角色所属菜单
      var menus=menuJavaService.findByUserId(userId);

      var js=new JSONObject()
      js.put("user",user)
      js.put("menus",menus)
      write(200,"查询成功","data",js)
    }catch {
      case e: Exception => {
        error(-1, e.getMessage)
      }
    }
  }

}
